### PR TITLE
EUI-2986: Task event completion

### PR DESCRIPTION
**Pull request by paul-graham**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-2986

### Change description ###
Initial work (for discussion for the time being) on adding support for completing work allocation tasks when case events are completed.

Note: Not a breaking change as the additional parameters passed around are currently optional.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/shared/components/case-viewer/case-event-trigger/case-event-trigger.component.ts [hunk 1]
```diff
@@ -29,9 +29,15 @@ export class CaseEventTriggerComponent implements OnInit, OnDestroy {
     private route: ActivatedRoute,
     private caseReferencePipe: CaseReferencePipe,
     private activityPollingService: ActivityPollingService
-  ) { }
+  ) {
+    // Doesn't look like this is ever called so this may be a redundant component now...
+    // It may, of course, just be that I don't know how to get to this component.
+    console.log('CaseEvenTriggerComponent constructor');
```
Comment (udaydenduluri33) [inline src/shared/components/case-viewer/case-event-trigger/case-event-trigger.component.ts:35]: please remove console.log

File: src/shared/components/case-viewer/case-event-trigger/case-event-trigger.component.ts [hunk 2]
```diff
@@ -29,9 +29,15 @@ export class CaseEventTriggerComponent implements OnInit, OnDestroy {
...
+  }
 
   ngOnInit(): void {
+    // Doesn't look like this is ever called so this may be a redundant component now...
+    console.log('CaseEvenTriggerComponent.ngOnInit', this.route.snapshot.data);
```
Comment (udaydenduluri33) [inline src/shared/components/case-viewer/case-event-trigger/case-event-trigger.component.ts:40]: please remove console.log

**Review by udaydenduluri33 — APPROVED** [review]
LGTMp

File: src/shared/components/case-viewer/case-event-trigger/case-event-trigger.component.ts (hunk 2 above)
Comment (paul-graham) [reply src/shared/components/case-viewer/case-event-trigger/case-event-trigger.component.ts:40]: This is a draft. All the console logs will go. There's a comment above it to explain why it's there. :)

File: src/shared/components/case-editor/services/cases.service.ts [hunk 3]
```diff
@@ -290,4 +294,53 @@ export class CasesService {
...
+          // If this fails, flag it as an error.
+        } else {
+          // TODO: This is a problem. We need to flag it up as an error.
+        }
+      }
+      // If we didn't get any tasks back, we're all good. Nothing to see here.
+    });
+  }
+
+  public getWorkAllocationTasks(taskSearchParameters: any[]): Observable<object> {
```
Comment (paul-graham) [inline src/shared/components/case-editor/services/cases.service.ts:330]: This is kinda the main bit that needs looking at. Is this in the right place? Have I understood the requirement correctly? Is the order in which it's processed correct (as in, have I called it from the right place)?

File: src/shared/components/case-editor/services/work-allocation.service.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      eventId,
+      state: [ 'Open' ] // Need to know which are the "completeable" statuses.
+    }];
+    return this.searchTasks({ parameters })
+      .pipe(
+        map((response: any) => {
+          const tasks: any[] = response.tasks;
+          if (tasks && tasks.length > 0) {
+            if (tasks.length === 1) {
+              this.completeTask(tasks[0].id).subscribe(() => console.log('Successfully processed tasks for this case event'));
```
[the remark this answers is not in the document]
Comment (udaydenduluri33) [reply src/shared/components/case-editor/services/work-allocation.service.ts]: removed it. Thanks.

**Review by rayliangatos — APPROVED** [review]

**Review by Mo-Lala84 — APPROVED** [review]
lgtm

**Pull request merged**